src/event_data.c

```diff
@@ -0,0 +1,325 @@
+            {
+                gLastQuestLogStoredFlagOrVarIdx = idx - VARS_START;
+                QuestLogSetFlagOrVar(0, idx, gSaveBlock1Ptr->vars[idx - VARS_START]);
+            }
+            break;
+        }
+        return &gSaveBlock1Ptr->vars[idx - VARS_START];
+    }
+    return gSpecialVars[idx - SPECIAL_VARS_START];
+}
+
+static bool8 IsFlagOrVarStoredInQuestLog(u16 idx, u8 a1)
+{
+    if (a1 == 0)
+    {
+        if (idx < STORY_FLAGS_START)
+            return FALSE;
+        if (idx >= SYS_FLAGS && idx < PERMA_SYS_FLAGS_START)
+            return FALSE;
+    }
+    else
+    {
+        if (idx < VAR_0x4030 - VARS_START)
+            return FALSE;
+        if (idx >= VAR_MAP_SCENE_PALLET_TOWN_OAK - VARS_START && idx < VAR_PORTHOLE - VARS_START)
+            return FALSE;
+    }
+    return TRUE;
+}
+
+u16 VarGet(u16 idx)
+{
+    u16 *ptr = GetVarPointer(idx);
+    if (ptr == NULL)
+        return idx;
+    return *ptr;
+}
+
+bool8 VarSet(u16 idx, u16 val)
+{
+    u16 *ptr = GetVarPointer(idx);
+    if (ptr == NULL)
+        return FALSE;
+    *ptr = val;
+    return TRUE;
+}
+
+u8 VarGetFieldObjectGraphicsId(u8 idx)
+{
+    return VarGet(VAR_OBJ_GFX_ID_0 + idx);
+}
+
+u8 *GetFlagAddr(u16 idx)
+{
+    u8 *ptr;
+    if (idx == 0)
+        return NULL;
+    if (idx < SPECIAL_FLAGS_START)
+    {
+        switch (gUnknown_3005E88)
+        {
+        case 0:
+        default:
+            break;
+        case 1:
+            ptr = QuestLogGetFlagOrVarPtr(1, idx);
+            if (ptr != NULL)
+                gSaveBlock1Ptr->flags[idx >> 3] = *ptr;
+            break;
+        case 2:
+            if (IsFlagOrVarStoredInQuestLog(idx, 0) == TRUE)
+            {
+                gLastQuestLogStoredFlagOrVarIdx = idx;
+                QuestLogSetFlagOrVar(1, idx, gSaveBlock1Ptr->flags[idx / 8]);
+            }
+            break;
+        }
+        return &gSaveBlock1Ptr->flags[idx / 8];
+    }
+    return &sSpecialFlags[(idx - SPECIAL_FLAGS_START) / 8];
+}
+
+bool8 FlagSet(u16 idx)
+{
+    u8 *ptr = GetFlagAddr(idx);
+    if (ptr != NULL)
+        *ptr |= 1 << (idx & 7);
+    return FALSE;
+}
+
+bool8 FlagClear(u16 idx)
+{
+    u8 *ptr = GetFlagAddr(idx);
+    if (ptr != NULL)
+        *ptr &= ~(1 << (idx & 7));
+    return FALSE;
+}
+
+bool8 FlagGet(u16 idx)
+{
+    u8 *ptr = GetFlagAddr(idx);
+    if (ptr == NULL)
+        return FALSE;
+    if (!(*ptr & 1 << (idx & 7)))
+        return FALSE;
+    return TRUE;
+}
+
+void ResetSpecialVars(void)
+{
+    gSpecialVar_0x8000 = 0;
+    gSpecialVar_0x8001 = 0;
+    gSpecialVar_0x8002 = 0;
+    gSpecialVar_0x8003 = 0;
+    gSpecialVar_0x8004 = 0;
+    gSpecialVar_0x8005 = 0;
+    gSpecialVar_0x8006 = 0;
+    gSpecialVar_0x8007 = 0;
+    gSpecialVar_0x8008 = 0;
+    gSpecialVar_0x8009 = 0;
+    gSpecialVar_0x800A = 0;
+    gSpecialVar_0x800B = 0;
+    gSpecialVar_Facing = 0;
+    gSpecialVar_Result = 0;
+    gSpecialVar_ItemId = 0;
```

Comment:
ITEM_NONE

Reply:
That would be inconsistent with local context.